File: src/Atlasd/Battlenet/Exceptions/ChatProtocolException.cs
using System;

namespace Atlasd.Battlenet.Exceptions
{
    class ChatProtocolException : ProtocolException
    {
        public ChatProtocolException(ClientState client) : base(Battlenet.ProtocolType.Types.Chat, client) { }
        public ChatProtocolException(ClientState client, string message) : base(Battlenet.ProtocolType.Types.Chat, client, message) { }
        public ChatProtocolException(ClientState client, string message, Exception innerException) : base(Battlenet.ProtocolType.Types.Chat, client, message, innerException) { }
    }
}

File: src/AtlasV/Battlenet/Exceptions/ChatProtocolException.vb
Imports System

Namespace AtlasV.Battlenet.Exceptions
    Class ChatProtocolException
        Inherits ProtocolException

        Public Sub New(ByVal varClient As ClientState)
            MyBase.New(Battlenet.ProtocolType.Types.Chat, varClient)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String)
            MyBase.New(Battlenet.ProtocolType.Types.Chat, varClient, varMessage)
        End Sub

        Public Sub New(ByVal varClient As ClientState, ByVal varMessage As String, ByVal varInnerException As Exception)
            MyBase.New(Battlenet.ProtocolType.Types.Chat, varClient, varMessage, varInnerException)
        End Sub
    End Class
End Namespace
